### PR TITLE
1542 ge config rework and hide selected elements

### LINKED ISSUE #1542: Hide selected elements via context menu/keyboard shortcut

### PR DESCRIPTION
**Pull request by philip-alldredge**
Closes #1542

### CODE REVIEW

**Comment by philip-alldredge** [comment]
@RyanMcilnay fixed the reported exception

File: ge/org.osate.ge/src/org/osate/ge/internal/ui/handlers/CreateDiagramHandler.java [hunk 1]
```diff
@@ -53,6 +53,7 @@ public void setEnabled(final Object evaluationContext) {
 		final Object contextBo = SelectionUtil.getDiagramContext(
 				AgeHandlerUtil.getCurrentSelection(),
 				AgeHandlerUtil.getActiveEditorFromContext(evaluationContext));
+
```
Comment (RyanMcilnay) [inline ge/org.osate.ge/src/org/osate/ge/internal/ui/handlers/CreateDiagramHandler.java]: Unnecessary change

File: ge/org.osate.ge/src/org/osate/ge/internal/ui/handlers/GoToPackageDiagramHandler.java [hunk 2]
```diff
@@ -11,20 +11,26 @@
...
-		final List<DiagramElement> selectedDiagramElements = AgeHandlerUtil
-				.getSelectedDiagramElements();
-		if (selectedDiagramElements.size() == 1) {
-			final Object selectedBo = selectedDiagramElements.get(0).getBusinessObject();
-			enabled = !(selectedBo instanceof AadlPackage) && getPackage(selectedBo) != null;
+		final List<BusinessObjectContext> selectedBusinessObjectContexts = AgeHandlerUtil
+				.getSelectedBusinessObjectContexts();
+		if (selectedBusinessObjectContexts.size() == 1) {
+			final Object selectedBo = selectedBusinessObjectContexts.get(0).getBusinessObject();
+			if(selectedBo instanceof AadlPackage) {
```
Comment (RyanMcilnay) [inline ge/org.osate.ge/src/org/osate/ge/internal/ui/handlers/GoToPackageDiagramHandler.java]: Seems like this could be cleaned up, example:
if(!(selectedBo instanceof AadlPackage)) {
    final AadlPackage pkg = getPackage(selectedBo);
    enabled = pkg != null && ProjectUtil.getProjectForBo(pkg).isPresent();
}

Unless there is an issue I am not seeing / or just preference.

File: ge/org.osate.ge/src/org/osate/ge/internal/ui/handlers/ShowContentsUtil.java [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+import org.osate.ge.internal.util.BusinessObjectProviderHelper;
+import org.osate.ge.services.ReferenceBuilderService;
+
+class ShowContentsUtil {
+	/**
+	 * Adds contents to the selected diagram elements. Adds all children which pass the specified filter.
+	 * @param event is the ExecutionEvent of the handler which provides the active editor.
+	 * @param filter is passed the parent diagram element and the business object of the potential child. If the filter returns true, then an element will be added for the business object.
+	 */
+	public static void addContentsToSelectedElements(ExecutionEvent event,
```
Comment (RyanMcilnay) [inline ge/org.osate.ge/src/org/osate/ge/internal/ui/handlers/ShowContentsUtil.java]: Add final to params

File: ge/org.osate.ge/src/org/osate/ge/internal/ui/handlers/ShowContentsUtil.java [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+
+	}
+
+	/**
+	 * Adds children of the specified diagram elements to the list of elements which will be added during the next diagram update.
+	 * @return whether children were added to the diagram.
+	 */
+	private static boolean addChildrenDuringNextUpdate(final List<DiagramElement> diagramElements, final DiagramUpdater diagramUpdater,
+			final ExtensionService extService, final ReferenceBuilderService referenceBuilder,
+			BiFunction<DiagramElement, Object, Boolean> filter) {
```
Comment (RyanMcilnay) [inline ge/org.osate.ge/src/org/osate/ge/internal/ui/handlers/ShowContentsUtil.java]: Add final to filter

File: ge/org.osate.ge/src/org/osate/ge/internal/ui/handlers/ShowElementHandler.java [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+		// Get services
+		final DiagramUpdater diagramUpdater = featureProvider.getDiagramUpdater();
+		final GraphitiService graphitiService = Objects.requireNonNull(
+				Adapters.adapt(diagramEditor, GraphitiService.class), "Unable to retrieve graphiti service");
+		final ReferenceBuilderService refBuilder = Objects.requireNonNull(
+				Adapters.adapt(diagramEditor, ReferenceBuilderService.class), "Unable to retrieve reference builder");
+		final ActionService actionService = Objects.requireNonNull(Adapters.adapt(diagramEditor, ActionService.class),
+				"Unable to retrieve action service");
+
+		BusinessObjectNode boTree = DiagramToBusinessObjectTreeConverter.createBusinessObjectNode(diagram);
```
Comment (RyanMcilnay) [inline ge/org.osate.ge/src/org/osate/ge/internal/ui/handlers/ShowElementHandler.java]: Add final

File: ge/org.osate.ge/src/org/osate/ge/internal/util/ProjectUtil.java [hunk 6]
```diff
@@ -47,4 +52,32 @@ public static IProject getProject(final URI elementUri) {
...
+
+	public static IProject getProjectForBoOrThrow(final Object bo) {
+		return getResource(bo).flatMap(r -> Optional.ofNullable(r.getURI())).flatMap(SelectionUtil::getProject).orElseThrow(() -> new RuntimeException("Unable to get project for business object: " + bo));
+	}
+
+	public static Optional<IProject> getProjectForBo(final Object bo) {
+		return getResource(bo).flatMap(r -> Optional.ofNullable(r.getURI())).flatMap(SelectionUtil::getProject);
+	}
+
+	private static Optional<Resource> getResource(Object bo) {
```
Comment (RyanMcilnay) [inline ge/org.osate.ge/src/org/osate/ge/internal/util/ProjectUtil.java]: add final to bo

File: ge/org.osate.ge/src/org/osate/ge/internal/businessObjectHandlers/SubprogramCallHandler.java [hunk 7]
```diff
@@ -87,40 +85,56 @@ public String validateName(final @Named(Names.BUSINESS_OBJECT) SubprogramCall ca
...
+				final SubprogramCall newSubprogramCall = cs.createOwnedSubprogramCall();
+				newSubprogramCall.setName(newSubprogramCallName);
+				newSubprogramCall.setContext(callContext);
+				newSubprogramCall.setCalledSubprogram(calledSubprogram);
+
+				AadlImportsUtil.ensurePackageIsImportedForClassifier(bi, callContext);
+				AadlImportsUtil.ensurePackageIsImportedForClassifier(bi, calledSubprogram);
+
+						// Build Result
+						StepResultBuilder<?> resultBuilder = StepResultBuilder.create().showNewBusinessObject(targetBoc,
```
[the remark this answers is not in the document]
Comment (philip-alldredge) [reply ge/org.osate.ge/src/org/osate/ge/internal/businessObjectHandlers/SubprogramCallHandler.java]: I believe I was able to fix this.

File: ge/org.osate.ge/src/org/osate/ge/internal/diagram/runtime/boTree/DefaultTreeUpdater.java [hunk 8]
```diff
@@ -313,29 +309,28 @@ private void createPropertyValues(final BusinessObjectNode node, final Property
...
 		final BusinessObjectNode oldNode = oldNodeMap.get(relReference);
 
 		// Create the node
-		final ImmutableSet<ContentFilter> contentFilters = oldNode == null || oldNode.getContentFilters() == null
+		final ImmutableSet<ContentFilter> contentFilters = oldNode == null
+				|| !oldNode.defaultChildrenHaveBeenPopulated()
 				? getDefaultContentFilters(diagramType, bo)
-						: oldNode.getContentFilters();
+						: ImmutableSet.of();
 				final UUID id = oldNode == null || oldNode.getId() == null ? UUID.randomUUID() : oldNode.getId();
```
[the remark this answers is not in the document]
Comment (philip-alldredge) [reply ge/org.osate.ge/src/org/osate/ge/internal/diagram/runtime/boTree/DefaultTreeUpdater.java]: Unable to get formatter to play nice. Going to leave the awkward formatting.

**Comment by philip-alldredge** [comment]
@RyanMcilnay All issues should be resolved.
